### DISCUSSION TITLE
Empty 'probabilityOfThunder'

### DISCUSSION THREAD

**Discussion by tflight** [weather-gov/api#328] [Q&A]
I'm working on a project where I'd like to look at data that forecasts the chance of thunderstorms. The `probabilityOfThunder` field would seem to be what I'm looking for, but [it appears empty](https://gist.github.com/tflight/7f80098e222191a1d5417dc57d76f848#file-gridpoints-gyx-62-123-json-L6354-L6356), even when in the `weather` it says [thunderstorms are likely](https://gist.github.com/tflight/7f80098e222191a1d5417dc57d76f848#file-gridpoints-gyx-62-123-json-L4309-L4319).

Am I misinterpreting what that field should show, or is there a reason it is empty that isn't obvious to me?

**Comment by mjgiancola** [comment]
I'm having a similar issue. I've looked at several locations and this field, as well as many others (probabilityOfTropicalStormWinds, probabilityOfHurricaneWinds, potentialOf15mphWinds, etc.) seem to always be empty. Are there any locations in which we could expect that these fields are given values?

**Reply by dwhitemv25** [reply]
The original post was from 2020. I'm sure OP is mightily confused why someone is responding to their 3 year old post.

The National Digital Forecast Database elements for tropical or convective hazards may only have data if data has been uploaded by the originating forecast center. If your weather doesn't meet the criteria then the value will be null or zero. Next time you think you should have data for an element, go to this site and check the desired elements (second dropbox from the left) and see if any non-zero probability covers your area:

https://digital.mdl.nws.noaa.gov/

(This is a graphical representation of the data source the `/gridpoints` endpoint uses.)

For the tropical elements, they may only show up if one of the forecast centers (NHC, SPC, WPC, etc.) has posted data for tropical weather.

**Comment by tflight** [comment]
> I'm sure OP is mightily confused why someone is responding to their 3 year old post.

No, I'm not confused why someone is responding to the post. I am still confused by the data. I was looking at this issue again about a month ago and still puzzled. The text forecast showed thunderstorms were likely yet `probabilityOfThunder` was empty.

**Reply by dwhitemv25** [reply]
I think the issue is one of the following.

1. The thunder probability in NDFD used to generate the text forecast isn't hooked up to the API. You can see the value in the hourly forecast grid on forecast.weather.gov. This value is set by the forecast office, and is manually adjusted to control the forecast text. (Forecasters here have mentioned bumping it up to get the thunder mention in, and I think the minimum threshold for mention is 20%-ish.)
2. NDFD has elements for Severe and Extreme Thunderstorm probability, provided by the Storm Prediction Center. That field might be connected to that data. It will only appear if the predicted conditions meet their thresholds.

Generalized thunderstorm probability can be found in various weather models. NBM has a field for it, which it's getting from GFS or a convective-allowing model.

**Reply by mjgiancola** [reply]
I didn't expect OP would still be interested, but clearly the issue is persisting, as it wasn't addressed 3 years ago when OP first identified it.

It seems to me to be the first one. I have yet to find a single instance where any of the following fields have a non-null value:

probabilityOfTropicalStormWinds
probabilityOfHurricaneWinds
potentialOf15mphWinds
potentialOf25mphWinds
potentialOf35mphWinds
potentialOf45mphWinds
potentialOf20mphWindGusts
potentialOf30mphWindGusts
potentialOf40mphWindGusts
potentialOf50mphWindGusts
potentialOf60mphWindGusts
grasslandFireDangerIndex
probabilityOfThunder

In particular, right now [https://digital.mdl.nws.noaa.gov/](https://digital.mdl.nws.noaa.gov/) is showing that Rawlins, WY (Lat: 41.69 N Lon: 107.31 W) is experiencing 30 mph wind gusts. However the field "potentialOf30mphWindGusts" is null: [https://api.weather.gov/gridpoints/CYS/26,46](https://api.weather.gov/gridpoints/CYS/26,46).

Is there a way to get this data from forecast.weather.gov in json format (or similar)?

**Reply by dwhitemv25** [reply]
I will have to go root around in the NDFD changelog for it, but I believe the "potential of X wind speed/gust" fields were discontinued. That or they come from NHC/SPC and will only show up for tropical weather (not likely to happen in Wyoming anytime soon).

Some of the forecast.weather.gov pages support a `FcstType=json` parameter to get a JSON version (or XML with `FcstType=dwml`). Not sure what level of support is offered for that interface, it's largely undocumented.

**Reply by mjgiancola** [reply]
Okay, Hilo, Hawaii then (Lat: 19.72 N Lon: 155.06 W), with a current "Hurricane Storm Surge Threat : Potential for storm surge > 1 ft above ground". All of the above fields are still null. So I guess all of these fields are discontinued.

**Reply by sullynole** [reply]
Are these fields being discovered in "forecastGridData" url from the api points endpoints?  This url contains raw gridded data from each office.  It is possible that some WFOs may have fields that are not available at other offices or are not always populated.  

While the NDFD website may start with some of the same data as the API, my understanding is that there may be some post-processing in NDFD to stitch together the fields for the digital.nws.noaa.gov maps, etc.  NDFD also contains some data from national offices which API does not, and NDFD does not show some fields from WFOs that are available in the raw gridded data in API.  

It may be best to reach out to a specific WFO if there are questions about their specific gridded elements in the raw data.

**Reply by mjgiancola** [reply]
Yes, these are fields in "forecastGridData". I understand that may be the case, but as I said above, I haven't found a _single_ instance in which _any_ of these fields are populated. I'm currently just trying to do simple experiments with the API so if anyone could point me to a case where literally any of the aforementioned fields are populated that would be super helpful.

**Reply by sullynole** [reply]
Here is a current example of probabilityOfThunder populated: https://api.weather.gov/gridpoints/TBW/62,91 
![image](https://github.com/weather-gov/api/assets/16312636/87d38e12-d05d-42d0-8fbc-909ea32c5aa9)

**Reply by tflight** [reply]
If there is no value, is there a way to determine if that is because there actually isn't a probability of thunder or if the data just isn't available for some reason?

**Reply by sullynole** [reply]
I am not sure how each office handles this field (and related fields).  My suggestion would be that missing is unknown either way.  Generally if it is there at a certain threshold, and depending on the PoP, then a chance of rain may become a chance of thunderstorms.

These are the kind of data that are used to create the forecast endpoints text and on forecast.weather.gov, so it's usually better to get that information there.

**Reply by mjgiancola** [reply]
I haven't been able to find this kind of data on forecast.weather.gov. The [MapClick API for Tampa Bay](https://forecast.weather.gov/MapClick.php?lat=27.4545&lon=-82.3245&unit=0&lg=english&FcstType=json) (the location from your example above) mentions thunderstorms in the forecasts but doesn't give the probabilities.

**Reply by ggocek** [reply]
In my experience, the property that I would like to display, but never has values, is "pressure", as in barometric pressure.
Grid Point -> properties -> pressure -> values
"values" is always empty. It seems like even the general public has an idea of the meaning of pressure and would be nice to see on a forecast.

**Reply by dwhitemv25** [reply]
@ggocek -- If you'd like to discuss this topic further, please start a new thread.
